jenkins/Jenkinsfile-blossom.premerge

```
@@ -685,3 +654,34 @@ String stashPytestResult(String name) {
 
     return result
 }
+
+void checkoutCode(String url, String sha) {
+    checkout(
+        changelog: false,
+        poll: true,
+        scm: [
+            $class           : 'GitSCM', branches: [[name: sha]],
+            submoduleCfg     : [],
+            userRemoteConfigs: [[
+                                    credentialsId: 'github-token',
+                                    url          : url,
+                                    refspec      : '+refs/pull/*/merge:refs/remotes/origin/pr/*']]
+        ]
+    )
+
+    stash(name: "source_tree", includes: "**,.git/**", useDefaultExcludes: false)
+}
+
+boolean databricksCodeChanged() {
+    def output = sh(script: '''
+            # get merge BASE from merged pull request. Log message e.g. "Merge HEAD into BASE"
+            BASE_REF=$(git --no-pager log --oneline -1 | awk '{ print $NF }')
+            git --no-pager diff --name-only ${BASE_REF} HEAD | grep -E 'sql-plugin/src/main/.*[0-9x-]db/|databricks || true'
```

The single-quotes on the grep argument are grabbing more than desired.  In addition we don't need the actual lines that matched, just whether or not something matched.
```suggestion
            git --no-pager diff --name-only ${BASE_REF} HEAD | grep -lE 'sql-plugin/src/main/.*[0-9x-]db/|databricks' || true
```

---

Good catch, thanks! I was intentionally to list the changed files so we could print out the matched lines (https://github.com/NVIDIA/spark-rapids/pull/5067/files#diff-6fcadc51af71fb51e6c978181ec240c5497d31a3af5c79d8baecaeea278b4fdaR683) to help developer understand why the CI got triggered w/ databricks build. But yeah, we may not need that until some debug scenarios actually occur.

I will refactor this part